server/src/main/java/org/apache/cassandra/sidecar/job/OperationsJobManager.java

```diff
@@ -0,0 +1,101 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.job;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Singleton;
+import org.apache.cassandra.sidecar.common.utils.OperationsJobResult.OperationsJobStatus;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+
+/**
+ * An abstraction of the management and tracking of long-running jobs running on the sidecar.
+ */
+@Singleton
+public class OperationsJobManager
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(OperationsJobManager.class);
+
+    private final OperationsJobTracker jobTracker;
+    private final ExecutorPools executorPools;
+
+    /**
+     * Creates a manager instance with a default sized job-tracker.
+     * @param executorPools
+     */
+    @Inject
+    public OperationsJobManager(ExecutorPools executorPools, OperationsJobTracker jobTracker)
+    {
+        this.executorPools = executorPools;
+        this.jobTracker = jobTracker;
+    }
+
+    /**
+     * Fetches the inflight jobs being tracked on the sidecar
+     * @return instances of the jobs that are in pending or running states
+     */
+    public List<OperationsJob> allInflightJobs()
+    {
+        return jobTracker.getJobsView().values()
+                         .stream()
+                         .filter(j -> j.status() == OperationsJobStatus.Pending || j.status() == OperationsJobStatus.Running)
+                         .collect(Collectors.toList());
+    }
+
+    /**
+     * Fetch the job using its UUID
+     * @param jobId identifier of the job
+     * @return instance of the job or null
+     */
+    public OperationsJob getJobIfExists(UUID jobId)
+    {
+        return jobTracker.get(jobId);
+    }
+
+    /**
+     * Asynchronously submit (and lazily create, via the supplier) the job, if it is not currently being
+     * tracked and is not running downstream. The job is triggered on a separate internal thread-pool.
+     * The job execution failure behavior is tracked within the {@link OperationsJob}.
+     * @param jobId job identifier
+     * @param jobSupplier supplier used to create an instance of the job
+     * @return the instance of the job that is either being tracked or was just submitted
+     */
+    public OperationsJob trySubmitJob(UUID jobId, Supplier<OperationsJob> jobSupplier)
```

Comment:
Passing a `Supplier` would have correctness issue. (Explained at the end) 
My suggestion is to have this signature instead.

```suggestion
public boolean trySubmitJob(OperationsJob job);
```

1. Use boolean to indicate the submission result.
2. `OperationsJob` already contains jobId. The parameter `UUID jobId` can be omitted.

Now, why there is potential correctness issue. 
Say there are 3 threads submitting the same jobId with different suppliers at the same time. Let's say thread A wins, thread B and C lose. Since the signature returns a `OperationsJob`, there is no way to tell which thread has succeeded. All A, B and C will consider they have created the job successfully. The ambiguity can break the subsequent execution.

Reply:
I changed this to use a creator function (`Function<UUID, OperationsJob> creator`) that takes UUID argument, to improve clarity on which ID is used for the job.  This should address the correctness issue as the returned job always corresponds to the headerId passed in (or a new UUID when absent).

As for the return-type, we need to job status and id to send the appropriate response.